datafusion/functions-array/src/macros.rs

```diff
@@ -106,4 +105,26 @@ macro_rules! make_udf_function {
             }
         }
     };
+    // This pattern does not generate the "fluent expr_fn" style function, it should be provided externally.
+    ($UDF:ty, $SCALAR_UDF_FN:ident) => {
```

Comment:
I would recommend to pull out `create_function` like https://github.com/apache/datafusion/blob/main/datafusion/functions-aggregate/src/macros.rs.

Reply:
Applied. This eliminates some duplication of macro codes. Thank you @jayzhan211 .